File: 2130-maximum-twin-sum-of-a-linked-list/2130-maximum-twin-sum-of-a-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public int pairSum(ListNode head) {
       
        ListNode i=head;
        int max=0;
        int length=1;
        while(i.next!=null){
            i=i.next;
            length++;
            
        }
        int[] arr=new int[length];
        int index=0;
        while(head.next!=null){
            arr[index++]=head.val;
            head=head.next;
        }
        arr[index]=head.val;
        int l=0;
        int r=length-1;
        System.out.println(arr[r]);
        while(l<r){
            int sum=arr[l]+arr[r];
            
            l++;
            r--;
            max=Math.max(max,sum);
        }
        return max;
    }
}

File: 2130-maximum-twin-sum-of-a-linked-list/2130-maximum-twin-sum-of-a-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def pairSum(self, head: Optional[ListNode]) -> int:
        arr=[]
        while head:
            arr.append(head.val)
            head=head.next
        l=0
        r=len(arr)-1
        m=0
        print(arr)
        while l<r:
            m=max(m,arr[l]+arr[r])
            l+=1
            r-=1
        return m